### PR TITLE
#149667055 Create contributions card interface

### PR DESCRIPTION
**Pull request by Mbarak-Mbigo**
#### What does this PR do?
Add contributions card interface
#### Description of Task to be completed?
Add contributions card interface
#### Any background context you want to provide?
This adds the interface for contribution cards
#### What are the relevant pivotal tracker stories?
[#149667055](https://www.pivotaltracker.com/story/show/149669999)
#### Screenshots (if appropriate)
screen shot 2017-08-01 at 11 27 43

### CODE REVIEW

**Comment by andela-kakpobome** [comment]
@Mbarak-Mbigo is this PR title formatted according to the convention we are adhering to?

File: src/pages/contributions/Contribution.scss [hunk 1]
```diff
@@ -1,26 +1,129 @@
...
-$card-main-width: 230px;
+  &:not(:last-child) {
+    margin-bottom: 0.5rem;
+  }
 
-#card-row {
-  /* Positioning */
-  float: $card-row-float;
+  .category {
+    background-color: #000;
```
Comment (andela-kakpobome) [inline src/pages/contributions/Contribution.scss]: #### Critical

We don't have one background color for all types of cards, you need to implement such that there are multiple sub-class-types for the different types of contributions that exist.

File: src/pages/contributions/Contribution.scss [hunk 2]
```diff
@@ -1,26 +1,129 @@
...
+
+  .vote-count, .comment-count i {
+    margin-right: 0.3rem;
+  }
+
+  .comment-count {
+    color: #c9c9c9;
+  }
+
+  .category-type {
```
Comment (andela-kakpobome) [inline src/pages/contributions/Contribution.scss]: ### Nitpick

Simplify this to ` .category-type i {`

File: src/pages/contributions/oldCard.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<Card className="card-main">
```
Comment (andela-kakpobome) [inline src/pages/contributions/oldCard.tsx]: ### Critical

This is now deprecated code and should be gotten rid off.

Comment (Mbarak-Mbigo) [inline src/pages/contributions/oldCard.tsx]: Okay

File: src/pages/contributions/Contribution.scss (hunk 1 above)
Comment (Mbarak-Mbigo) [inline src/pages/contributions/Contribution.scss]: fixed

**Comment by Mbarak-Mbigo** [comment]
@andela-kakpobome I have fixed the issues

File: src/pages/contributions/ContributionPage.component.tsx [hunk 4]
```diff
@@ -54,7 +54,9 @@ interfaces.IContributionsPageState> {
   public render() {
     const contributions = this.state.contributions;
     return (
-      <div><ContributionList contributions={contributions}/></div>
+      <div>
+        <ContributionList contributions={contributions}/>
```
Comment (BolajiOlajide) [inline src/pages/contributions/ContributionPage.component.tsx]: I think you should space out the content of the curly brace.
`<ContributionList contributions={ contributions } />`

File: src/pages/contributions/ContributionsCard.tsx [hunk 5]
```diff
@@ -20,25 +20,68 @@ const checkCategory = (contributionsCat) => {
 export const ContributionsCard = ({ contributions }) => {
   checkCategory(contributions.Category);
 
+  let category = 'category';
+  if (contributions.Category === 'Skills') {
```
Comment (andela-kakpobome) [inline src/pages/contributions/ContributionsCard.tsx]: ### Critical

1. Modify the `Category` property in the mocks to be `lower case` so this becomes
`contributions.category`, as this is how the actual endpoint's data will look like.

2. Change the values to either one of `skill`, `resource`, `idea`.

3.  Abstain from `concatenation` use `template strings` instead [See here](https://github.com/airbnb/javascript#es6-template-literals)

4. Look [here](https://github.com/airbnb/javascript#whitespace--around-keywords) to see how to properly format conditionals and other control blocks

5. Replace the `checkCategory` method's content with this conditional evaluation and rename it to `getCategoryClass`

**Comment by andela-kakpobome** [comment]
@Andela-Didacus Shouldn't you be the one reviewing these PRs this week before I do?

File: src/pages/contributions/ContributionPage.component.tsx (hunk 4 above)
Comment (Andela-Didacus) [inline src/pages/contributions/ContributionPage.component.tsx]: @BolajiOlajide what he had initially is correct, we only space it out during import i.e `import { Contributions } from .......`

Comment (BolajiOlajide) [inline src/pages/contributions/ContributionPage.component.tsx]: Noted.

File: src/pages/contributions/Contribution.scss [hunk 6]
```diff
@@ -1,26 +1,143 @@
...
-$card-title-height: 100px;
+.contribution-card {
+  display: flex;
+  width: 100%;
+  height: 10rem;
+  -webkit-box-sizing: border-box;
+  -moz-box-sizing: border-box;
+  box-sizing: border-box;
+  border-radius: 0.25rem;
+  border: 1px solid #d8d8d8;
```
Comment (andela-kakpobome) [inline src/pages/contributions/Contribution.scss]: ### Critical

1. Change `ALL` these colors to read directly from the value maps in our `theme.scss`.

2. In the `theme.scss`, import the `font.scss`(as the first import) and set up our text elements to use the primary font, so i.e:

```
body, h1, h2, h3, h4, p, a {
   font-family: xxxx
}
```

File: src/pages/contributions/ContributionsCard.tsx [hunk 7]
```diff
@@ -1,44 +1,67 @@
...
+      <div className={getCategoryClass(contributions.category)}>
+        <div className="neutral-state">
+          <div className="category-type">
+            <i className="large material-icons">lightbulb_outlines</i>
+          </div>
+        </div>
+
+        <div className="hover-state">
+          <div className="category-type">
+            <i className="large material-icons">lightbulb_outlines</i>
```
Comment (andela-kakpobome) [inline src/pages/contributions/ContributionsCard.tsx]: ### Critical

As Didah mentioned, you also need to make the icon dynamic and based on the category.

File: src/pages/contributions/Contribution.scss [hunk 8]
```diff
@@ -1,21 +1,167 @@
...
+
+    .value {
+      font-size: 0.9rem;
+    }
+  }
+
+  .details {
+    position: relative;
+    flex: 1;
+    padding: 0.5rem;
```
Comment (andela-kakpobome) [inline src/pages/contributions/Contribution.scss]: ### Nitpick

I noticed we need more padding on the sides, so let this be `0.5rem 1rem`

File: src/pages/contributions/Contribution.scss [hunk 9]
```diff
@@ -1,21 +1,167 @@
...
+      font-size: 0.85rem;
+      color: map_get($grey, 400);
+    }
+
+    .tags {
+      margin-top: 1rem;
+    }
+
+    .tag {
+      color: map_get($grey, 400);
```
Comment (andela-kakpobome) [inline src/pages/contributions/Contribution.scss]: ### Critical

It's currently still too big, set the font-size to `0.5rem`

File: src/pages/contributions/Contribution.scss [hunk 10]
```diff
@@ -1,21 +1,167 @@
...
+      }
+    }
+  }
+
+  .vote-count, .comment-count, .category-type {
+    display: flex;
+    align-items: center;
+  }
+
+  .vote-count, .comment-count i {
```
Comment (andela-kakpobome) [inline src/pages/contributions/Contribution.scss]: Set font-size to 1rem, and make sure everything inside these containers are vertically aligned.

File: src/pages/contributions/Contribution.scss [hunk 11]
```diff
@@ -1,21 +1,167 @@
...
+    flex: 1;
+    padding: 0.5rem;
+
+    .bookmark {
+      position: absolute;
+      top: 0.5rem;
+      right: 1rem;
+    }
+
+    .title {
```
Comment (andela-kakpobome) [inline src/pages/contributions/Contribution.scss]: This should have a `font-weight: 400`, based on the design

Comment (Mbarak-Mbigo) [inline src/pages/contributions/Contribution.scss]: 👍

File: src/pages/contributions/Contribution.scss (hunk 8 above)
Comment (Mbarak-Mbigo) [inline src/pages/contributions/Contribution.scss]: 👍

File: src/pages/contributions/Contribution.scss (hunk 10 above)
Comment (Mbarak-Mbigo) [inline src/pages/contributions/Contribution.scss]: 👍

File: src/pages/contributions/ContributionsCard.tsx [hunk 12]
```diff
@@ -1,44 +1,97 @@
...
+ *
+ * @param category
+ * @returns {string} material iconName
+ */
+const getCategoryIcon = (category) => {
+  if (category === 'skill') {
+    return `all_inclusive`;
+  }else if (category === 'resource') {
+    return `attach_file`;
+  }else {
```
Comment (andela-kakpobome) [inline src/pages/contributions/ContributionsCard.tsx]: ### Critical

I've given feedback about how to space out blocks, why is this still like this?

File: webpack.config.prod.js [hunk 13]
```diff
@@ -19,8 +19,8 @@ module.exports = {
   module: {
     loaders: [
       { test: /\.tsx?$/, include: path.join(__dirname, 'src'), loaders: ['ts-loader', 'tslint-loader'] },
-      { test: /(\.css)$/, loader: ExtractTextPlugin.extract('css-loader?sourceMap&modules&importLoaders=1&localIdentName=[name]__[local]___[hash:base64:5]!postcss-loader!sass-loader') },
-      { test: /(\.scss)$/, loaders: ['style-loader', 'css-loader','sass-loader'] },
+      { test: /(\.css)$/, loader: ExtractTextPlugin.extract('scss-loader?sourceMap&modules&importLoaders=1&localIdentName=[name]__[local]___[hash:base64:5]!postcss-loader!sass-loader') },
+      { test: /(\.scss)$/, loaders: ['style-loader', 'scss-loader','sass-loader'] },
```
Comment (andela-kakpobome) [inline webpack.config.prod.js]: ### Critical

Why is this scss loader still here after I've given feedback and you observed that it was wrong?

**Comment by Mbarak-Mbigo** [comment]
I do understand.

**Pull request merged**